spacepy/pybats/pwom.py

```diff
@@ -142,6 +149,146 @@ def _read(self, starttime):
                 for k,v in enumerate(var):
                     self[v][i,j]=float(parts[k+1])
 
+    def _read_binary(self, starttime):
+        """Read binary line file; should only be called upon instantiation.
+
+        Parameters
+        ----------
+        starttime : datetime.datetime
+            Start date and time of the simulation
+        """
+        char_size = 1
+        int_size = 4
+        long_size = 8
+        float_size = 4
+        dbl_size = 8
+        # Slurp whole file.
+        fileContent = open(self.attrs['file'], 'rb').read()
+
+        #get the record length
+        RecordLength=struct.unpack('i', fileContent[0 : 0+int_size])
+        #save the first time
+        NextDataStart = RecordLength[0] + 2*int_size + 2*int_size
+        t = struct.unpack('d', fileContent[NextDataStart : NextDataStart+dbl_size])
+
+        #next data skiping ndim and nparam
+        NextDataStart = NextDataStart + dbl_size + int_size + int_size
+
+        #get number of vars
+        nVar = struct.unpack('i',
+                             fileContent[NextDataStart : NextDataStart+int_size])
+        nVars = nVar[0]
+        NextDataStart = NextDataStart + 3*int_size
+        #get nAlts
+        nAlt = \
+            struct.unpack('i',
+            fileContent[NextDataStart : NextDataStart+int_size])
+        nAlts = nAlt[0]
+        NextDataStart = NextDataStart + 3*int_size
+        #skip reading params
+        NextDataStart = NextDataStart + 2*dbl_size
+
+        #read vars
+        NameVarList = \
+            str(struct.unpack('77s',
+            fileContent[NextDataStart : NextDataStart+77]))
+        # Get variable names; pop radius (altitude).
+        var = NameVarList.split()[1:-1]
+        #skip to next record
+        i = NextDataStart + 77
+        while fileContent[i : i+1] == b' ':
+            i += 1
+        NextDataStart = i + 8
+
+        #get the total length of a snapshot
+        #note the 164 comes from the byte pads between variables
+        SnapshotLength = NextDataStart + nAlts*dbl_size*(nVars+1) + 164
+        #get number of times in the file
+        nTimes = int(len(fileContent) / SnapshotLength)
+        #correct length of snapshot due to padding issues
+        #SnapshotLength=int(len(fileContent)/nTimes)
```

Comment:
I removed a few other commented-out debugging print statements, but wasn't sure about this. If it's necessary we should uncomment it; otherwise probably should go. @aglocer ?